src/main/scala/org/constellation/gossip/state/GossipMessagePathTracker.scala

```diff
@@ -0,0 +1,59 @@
+package org.constellation.gossip.state
+
+import cats.effect.Concurrent
+import cats.effect.concurrent.Ref
+import cats.implicits._
+
+class GossipMessagePathTracker[F[_]: Concurrent, A] {
+  private val paths: Ref[F, Map[String, GossipMessageState]] = Ref.unsafe(Map.empty)
+
+  def start(message: GossipMessage[A]): F[Unit] =
+    paths.modify { m =>
+      (m + (message.path.id -> GossipMessageState.Pending), ())
```

Comment:
We could actually get this working (with the changes to the fanout fcn and removing timeout) by making the message.path.id be the proposal hash with the prev node's id and height concatenated in some way. That should be good enough

Reply:
>message.path.id be the proposal hash with the prev node's id and height concatenated in some way

Sure, we can use proposal hash as uuid. Of course if we assume that `height -> hash` proposal is sent only once.
There are some rare cases when same `height -> hash` proposal is sent multiple times but via different partition. Then we would like to distinguish partitions from each other.

I guess such edge case can happen if node makes snapshot at specific height and then redownload drops it down because it went too high according to majority. Then it technically can create same snapshot with same height and hash again. 🤔

Reply:
But in general I like the idea of using the data itself to provide uniqueness instead of relying on random uuid strings. I am just a little bit worries about uniqueness of `height+hash+prevId` in the context of redownloads.

Reply:
Maybe we should just filter such proposal out at the moment of sending 🤔 
Proposals are immutable so it shouldn't send specific height multiple times when redownload drops node down to lower height.